### PR TITLE
Add LayoutLMv2 OnnxConfig

### PR DESCRIPTION
**Pull request by chainyo**
# What does this PR do?

Add LayoutLMv2 OnnxConfig to make this model available for conversion.

I took the same config as `LayoutLM` and added the adapted shebang.

## Who can review?

Models: @LysandreJik @lewtun

### CODE REVIEW

**Comment by NielsRogge** [comment]
There's quite some interest for ONNX support of LayoutLMv2, see #14368 and #14555.

**Comment by ChainYo** [comment]
> There's quite some interest for ONNX support of LayoutLMv2, see #14368 and #14555.

Ok thanks for the links, I will look at them when I got some time and start improving the actual PR (it seems that the last PR has been closed due to inactivity, so I will take what I can from this and improve my own PR)

Thanks!

File: src/transformers/onnx/config.py [hunk 1]
```diff
@@ -235,6 +235,22 @@ def _generate_dummy_images(
             images.append(Image.fromarray(data.astype("uint8")).convert("RGB"))
         return images
 
+    def _generate_dummy_bbox(self, batch_size: int = 2, image_height: int = 40, image_width: int = 40) -> List[int]:
```
Comment (ChainYo) [inline src/transformers/onnx/config.py:238]: I have added a way to generate `dummy_bbox` if required by the preprocessor. The bounding box are scaled on the image size by default.

File: src/transformers/onnx/config.py [hunk 2]
```diff
@@ -295,6 +314,10 @@ def generate_dummy_inputs(
             )
             # Generate dummy inputs according to compute batch and sequence
             dummy_input = [" ".join([preprocessor.unk_token]) * seq_length] * batch_size
+            # Generate dummy bounding boxes if needed by the preprocessor e.g. for LayoutLMv2
+            if return_bbox is True:
```
Comment (ChainYo) [inline src/transformers/onnx/config.py:318]: If the `return_bbox` is True then it means the preprocessor needs bounding boxes and it returns the appropriate dictionary with dummy_bbox.

**Comment by jhubar** [comment]
@ChainYo have you find a solution for tensors problem ? 🤗

**Comment by ChainYo** [comment]
> @ChainYo have you find a solution for tensors problem ? hugs

It seems that one tensor size is changing for no reason on my previous tests. I will dig more this week, I had no time last week.

**Comment by jhubar** [comment]
Great,if you want, we can organise a Google meet and looking the issue together. You can send me an email at : hubarjulien@gmail.com

**Comment by sujit420** [comment]
@ChainYo I am getting errors while using your PR during inference time. I am using the below code for token-classification (FUNSD):
processor = LayoutLMv2Processor.from_pretrained("microsoft/layoutlmv2-base-uncased", revision="no_ocr")
image_path = '../funsd/page1.png'
image = Image.open(image_path).convert("RGB")
words, bboxes = get_words_and_boxes_textract(textract_client, image_path)
encoded_inputs = processor(image, words, boxes=bboxes, padding="max_length", truncation=True, return_tensors="pt")
for k,v in encoded_inputs.items():
encoded_inputs[k] = v.to(device)
dt = datasets.Dataset.from_dict(encoded_inputs)
outputs = loaded_ort_model.evaluation_loop(dt)

Error:
InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Invalid Feed Input Name:token_type_ids

Any help is really appreciated.

**Comment by jhubar** [comment]
You can try to add to the processor function token_type_ids like this :
```
  input_ids = batch['input_ids'].to(device)
  bbox = batch['bbox'].to(device)
  image = batch['image'].to(device)
  attention_mask = batch['attention_mask'].to(device)
  token_type_ids = batch['token_type_ids'].to(device)
  labels = batch['labels'].to(device)
  # forward pass
   outputs = model(input_ids=input_ids, bbox=bbox, image=image, attention_mask=attention_mask, 
                            token_type_ids=token_type_ids, labels=labels)
```

**Comment by sujit420** [comment]
> outputs = model(input_ids=input_ids, bbox=bbox, image=image, attention_mask=attention_mask, 
>                             token_type_ids=token_type_ids, labels=labels)

there is a method called evaluation_loop which does inferencing for loaded onnx models. it expects only huggingface dataset as per https://github.com/huggingface/optimum.

Please look at the snippet which I posted earlier.

**Comment by malcolmgreaves** [comment]
Hi! Thanks for working on this @ChainYo 🙏 I am interested in this work. Are you still working on this?

**Comment by lewtun** [comment]
Hey @ChainYo, regarding your error with the tests - my guess is that the dummy data generation is the culprit. My suggestion would be to:

* First pick an input that works with the `torch` model
* Export the model to ONNX and check the forward pass still works with the same input
* Generalise to the dummy input case

**Comment by ChainYo** [comment]
@malcolmgreaves I don't even remember where I was with this issue months ago. But I will try to work on it this week if I can.

**Comment by ChainYo** [comment]
@lewtun You are right! It seems that someone solved the things I was trying to achieve by doing it with LayoutLMv3. So I will check that and see what I can apply to v2.

**Comment by vibhas-singh** [comment]
Hey @ChainYo 
Were you able to check this?
Any direction on this would be really helpful.

> @lewtun You are right! It seems that someone solved the things I was trying to achieve by doing it with LayoutLMv3. So I will check that and see what I can apply to v2.
